packages/artifact/src/internal/shared/artifact-twirp-client.ts

```diff
@@ -84,11 +85,11 @@ class ArtifactHttpClient implements Rpc {
         debug(`[Response] - ${response.message.statusCode}`)
         debug(`Headers: ${JSON.stringify(response.message.headers, null, 2)}`)
         debug(`Body: ${body}`)
-
         if (this.isSuccessStatusCode(statusCode)) {
           return {response, body}
         }
-
+        // only parse the JSON if status is non-successful
+        errorResponse = JSON.parse(body).msg
```

Comment:
meganit: let's reuse the existing `errorMessage`

move this down past line 94 and have something like:

```js
const responseMessage = JSON.parse(body).msg
if (responseMessage) {
  errorMessage = `${errorMessage}: ${responseMessage}`
}
```